utils/generic-components.stanza

```diff
@@ -173,19 +210,28 @@ public defn ceramic-cap (user-params:Tuple<KeyValue>) -> Instantiable :
   val params = remove-duplicate-keys $ [["type" => "ceramic" "min-rated-voltage" => MIN-CAP-VOLTAGE]
                                         smd-query-properties()
                                         user-params]
-  to-jitx $ Capacitor(params)
-
-public defn ceramic-cap (cap:Double) -> Instantiable :
+  val my-params = replace-params(params)
+  try :
+    to-jitx $ Capacitor(my-params)
+  catch (e) :
+    val my-cap = lookup-val(params, "capacitance")
+    println("Warning: generic capacitor substitution for value (%_)" % [my-cap])
+    match(my-cap) :
+      (d:Double) : dummy-capacitor(d)
+      (t:Toleranced) : dummy-capacitor(t)
+      (_) : fatal("Did not find acceptable type of capacitance value (%_)" % [my-cap])
```

Comment:
Do you want to specifically catch `jitx/errors/NoComponentMeetingRequirements` exceptions?
Errors from the database can be of:
- NoComponentMeetingRequirements
- InvalidPartDatabaseQueryAttributeType
- InvalidPartDatabaseQuerySpecialKey
- InvalidPartDatabaseQueryProperty
- HTTPError

Reply:
I also want to catch the other errors you listed because we don't want the design to crash. We print a warning when we use a dummy-* component so that the user can continue compiling the design even if no internet connectivity. With the warning, the user can go back and modify their code if they want real components (fix query, wait for internet connectivity).

Reply:
Ok, however you are silently catching the error and printing a hardcoded different (potentially wrong) error message to the terminal.
You need to still print the error.
The typo checker has elaborate error printers listing the available attributes and special keys when the user makes a typo.
The typo checker is actually meant to stop execution and let the user fix the query.

Reply:
Ok, will now only catch HTTPError & NoComponentMeetingRequirements. The remainder will trigger the in-place error handlers.